java/udf/src/test/java/com/risingwave/functions/TestUdfServer.java

```diff
@@ -82,28 +86,133 @@ public void gcd() throws Exception {
+                BigDecimal decimal,
+                LocalDate date, LocalTime time, LocalDateTime timestamp, PeriodDuration interval,
+                String str, byte[] bytes, @DataTypeHint("JSONB") String jsonb) {
+            var row = new Row();
+            row.bool = bool;
+            row.i16 = i16;
+            row.i32 = i32;
+            row.i64 = i64;
+            row.f32 = f32;
+            row.f64 = f64;
+            row.decimal = decimal;
+            row.date = date;
+            row.time = time;
+            row.timestamp = timestamp;
+            row.interval = interval;
+            row.str = str;
+            row.bytes = bytes;
+            row.jsonb = jsonb;
+            return row;
         }
     }
 
     @Test
-    public void to_string() throws Exception {
-        var c0 = new VarCharVector("", allocator);
-        c0.allocateNew(1);
-        c0.set(0, "string".getBytes());
-        c0.setValueCount(1);
-        var input = VectorSchemaRoot.of(c0);
-
-        try (var stream = client.call("to_string", input)) {
+    public void all_types() throws Exception {
+        var c0 = new BitVector("", allocator);
+        c0.allocateNew(2);
+        c0.set(0, 1);
+        c0.setValueCount(2);
+
+        var c1 = new SmallIntVector("", allocator);
+        c1.allocateNew(2);
+        c1.set(0, 1);
+        c1.setValueCount(2);
+
+        var c2 = new IntVector("", allocator);
+        c2.allocateNew(2);
+        c2.set(0, 1);
+        c2.setValueCount(2);
+
+        var c3 = new BigIntVector("", allocator);
+        c3.allocateNew(2);
+        c3.set(0, 1);
+        c3.setValueCount(2);
+
+        var c4 = new Float4Vector("", allocator);
+        c4.allocateNew(2);
+        c4.set(0, 1);
+        c4.setValueCount(2);
+
+        var c5 = new Float8Vector("", allocator);
+        c5.allocateNew(2);
+        c5.set(0, 1);
+        c5.setValueCount(2);
+
+        var c6 = new DecimalVector("", allocator, 38, 0);
+        c6.allocateNew(2);
+        c6.set(0, BigDecimal.valueOf(10).pow(37));
+        c6.setValueCount(2);
+
+        var c7 = new DateDayVector("", allocator);
+        c7.allocateNew(2);
+        c7.set(0, (int) LocalDate.of(2023, 1, 1).toEpochDay());
+        c7.setValueCount(2);
+
+        var c8 = new TimeMicroVector("", allocator);
+        c8.allocateNew(2);
+        c8.set(0, LocalTime.of(1, 2, 3).toNanoOfDay() / 1000);
+        c8.setValueCount(2);
+
+        var c9 = new TimeStampMicroVector("", allocator);
+        c9.allocateNew(2);
+        var ts = LocalDateTime.of(2023, 1, 1, 1, 2, 3);
+        c9.set(0, ts.toLocalDate().toEpochDay() * 24 * 3600 * 1000000 + ts.toLocalTime().toNanoOfDay() / 1000);
+        c9.setValueCount(2);
+
+        var c10 = new IntervalMonthDayNanoVector("", FieldType.nullable(MinorType.INTERVALMONTHDAYNANO.getType()),
+                allocator);
+        c10.allocateNew(2);
+        c10.set(0, 1, 2, 3);
```

Comment:
nit: maybe test with 3 microsecond rather than 3 nanosecond, which does not happen in real cases.